internal/server/admin/keyset_service.go

```diff
@@ -0,0 +1,144 @@
+package admin
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	pbModel "github.com/kong/koko/internal/gen/grpc/kong/admin/model/v1"
+	v1 "github.com/kong/koko/internal/gen/grpc/kong/admin/service/v1"
+	"github.com/kong/koko/internal/model"
+	"github.com/kong/koko/internal/resource"
+	"github.com/kong/koko/internal/server/util"
+	"github.com/kong/koko/internal/store"
+	"go.uber.org/zap"
+)
+
+type KeySetService struct {
+	v1.UnimplementedKeySetServiceServer
+	CommonOpts
+}
+
+func (s *KeySetService) GetKeySet(
+	ctx context.Context,
+	req *v1.GetKeySetRequest,
+) (*v1.GetKeySetResponse, error) {
+	logger := s.logger(ctx)
```

Comment:
Can we validate the presence of the ID for consistency with the other entities?

```
        if req.Id == "" {
		return nil, s.err(ctx, util.ErrClient{Message: "required ID is missing"})
	}
```

Reply:
done

Reply:
You will need to fix some tests like [this](https://github.com/Kong/koko/blob/6888d9f698f7416d7bd18e36ef2f56fdcaff333f/internal/server/admin/key_test.go#L108) after the change